src/details/ArborX_DetailsLegacy.hpp

```diff
@@ -61,27 +61,29 @@ struct LegacyCallbackWrapper
 {
   Callback _callback;
 
-  template <typename Predicate, typename Value>
-  KOKKOS_FUNCTION auto operator()(Predicate const &predicate,
-                                  PairValueIndex<Value> const &value) const
+  template <typename Predicate, typename Value, typename Index>
+  KOKKOS_FUNCTION auto
+  operator()(Predicate const &predicate,
+             PairValueIndex<Value, Index> const &value) const
   {
     return _callback(predicate, value.index);
   }
 
-  template <typename Predicate, typename Value, typename Output>
-  KOKKOS_FUNCTION auto operator()(Predicate const &predicate,
-                                  PairValueIndex<Value> const &value,
+  template <typename Predicate, typename Value, typename Index, typename Output>
+  KOKKOS_FUNCTION void operator()(Predicate const &predicate,
```

Comment:
Why did you change the return type?

Reply:
For clarity. Our 3-arg callback is only allowed to return void`.

Reply:
Do you remember why we required it to return `void`?

Reply:
Because the semantics of early termination when doing two pass are unclear.